### PR TITLE
Analysis Wavelet Transform

### PR DESCRIPTION
**Pull request by JessWhitney**
Adds the **analysis transform** (corresponding to issue #11) to s2wav. The original C code used for reference can be found [here](https://github.com/astro-informatics/s2let/blob/main/src/main/c/s2let_analysis.c).

- Adds module to the init file
- Adds core code to 'analysis.py'
- Doesn't include any 'manual' or 'real' functions, only the core transform

### CODE REVIEW

File: s2wav/analysis.py [hunk 1]
```diff
@@ -51,48 +53,71 @@ def analysis_transform(flm: np.ndarray,
...
+        for n in range(-N + 1, N, 2):
+            for el in range(max(abs(spin), abs(n)), L):
                 if el != 0:
-                    psi = 8 * np.pi * np.pi / (2 * el + 1) * np.conj(wav_lm[j, el, L - 1 + n])
-                    for m in range (-el, el + 1):
-                        f_wav_lmn[j, N - 1 + n, el, L - 1 + m] = flm[el, L - 1 + m] * psi
-
-    for el in range(abs(spin), L +1):
+                    psi = np.conj(wav_lm[j, el, L - 1 + n])
+                    psi *= 8 * np.pi**2 / (2 * el + 1)
```
Comment (JessWhitney) [inline s2wav/analysis.py:68]: @CosmoMatt Are you splitting the psi calculation over 2 lines so it's more readable or is there another reason?

Comment (CosmoMatt) [reply s2wav/analysis.py:68]: Nope, just so it's readable. Feel free to switch back if you prefer

Comment (JessWhitney) [reply s2wav/analysis.py:68]: Will keep it as is, think the way you've written it is nicer.

**Comment by JessWhitney** [comment]
> 

Would you like me to wait for @alicjapolanska to take a look before merging, or merge now?

**Pull request merged**